File: Python/Relative Ranks.py
class Solution(object):
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        lst = sorted(nums, reverse=True)
        rank = ["Gold Medal", "Silver Medal", "Bronze Medal"] + [str(num) for num in range(4, len(nums) + 1)]
        bind_dict = dict(zip(lst, rank))
        return [bind_dict[i] for i in nums]

File: cpp/Relative Ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& nums) {
        vector<int> lst = nums;
        vector<string> rank;
        sort(lst.begin(), lst.end(), greater_equal<int>());
        rank.push_back(string("Gold Medal"));
        rank.push_back(string("Silver Medal"));
        rank.push_back(string("Bronze Medal"));
        for (int i = 4; i < nums.size() + 1; ++i) {
            rank.push_back(to_string(i));
        }
        map<int, string> temp;
        for (int j = 0; j < lst.size(); ++j) {
            temp[lst[j]] = rank[j];
        }
        vector<string> result;
        for (int num : nums) {
            result.push_back(temp[num]);
        }
        return result;
    }
};
